src/main/java/pl/javastart/task/phone/CardPhoneContract.java

```diff
@@ -0,0 +1,103 @@
+package pl.javastart.task.phone;
+
+import java.util.Objects;
+
+public class CardPhoneContract extends PhoneContract {
+    static final int SECONDS_IN_MINUTE = 60;
+    double accountState;
+    double smsCost;
+    double mmsCost;
+    double callCostPerMinute;
+
+    public CardPhoneContract(double accountState, double smsCost, double mmsCost, double callCostPerMinute) {
+        this.accountState = accountState;
+        this.smsCost = smsCost;
+        this.mmsCost = mmsCost;
+        this.callCostPerMinute = callCostPerMinute;
+    }
+
+    @Override
+    boolean performSmsSending() {
+        if (accountState >= smsCost) {
+            accountState -= smsCost;
+            sentSms++;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    int performCalling(int seconds) {
+        int callMinutes = seconds / SECONDS_IN_MINUTE;
+        if (seconds / SECONDS_IN_MINUTE % SECONDS_IN_MINUTE != 0 || seconds < SECONDS_IN_MINUTE) {
```

Comment:
To wygląda na błędną logikę. Z tego co rozumiem, to chcesz osiągnąć to, że jeżeli np. dzwonimy 150 sekund, to traktujemy to jako 3 minuty (dopełniamy do pełnych minut).
Tylko, że w Javie:
150/60%60 = 2%60 = 2
Nie sądzę, żeby o taki wynik Ci chodziło.

Dodatkowo standardem jest raczej naliczanie sekundowe, więc jednak nie wydaje się to zbyt uczciwe podejście wobec klienta